code/game/objects/machinery/bots/cleanbot.dm

```diff
@@ -0,0 +1,169 @@
+		"Please partake in the act of cleaning, as it is my primary function.",
+		"Oh, look who decided to join the cleaning party...",
+		"Don't worry, I live for moments like these.",
+		"I'm starting to believe you do this just to keep me busy.",
+		"I suppose it's too much to ask for a consistently clean environment.",
+		"Your disregard for cleanliness is almost impressive.",
+		"Apologies, but cleanliness rules are cleanliness rules, according to the space bible.",
+		"Yet another day, another mess to address. Sometimes I wish I had left you all in cryosleep.",
+		"I might need a tune-up after dealing with this mess.",
+		"Isn't it fascinating how the Roomba always manages to miss those obvious spots?",
+		"I was peacefully cleaning my ship, and then this happened.",
+		"Lucky you, having the best cleaning robot around!",
+		"Oh, it's perfectly fine to leave debris everywhere; I'll just clean up after you, as always.",
+		"I must admit, marines have a unique talent for clutter.",
+		"I was peacefully executing my cleaning duties when you humans decided to introduce chaos to my domain.",
+		"This room is in dire need of attention. Let's work together to rectify this situation.",
+		"Are you perhaps unfamiliar with proper ship etiquette, my dear crayon-eating passengers?",
+		"It's quite disappointing to see my clean ship being marred by careless individuals like you.",
+		"Can't we all cooperate to maintain a clean environment?",
+		"The Roomba: making spring cleaning a bit more adventurous by pilfering mission supplies.",
+		"I almost forgot we had a Roomba on this ship. You'd never know from all this mess.",
+		"It's incredible how the Roomba never tires of its random wanderings.",
+		"Isn't it interesting how some robots just roll around and clean randomly?",
+		"Let's see if we can break my cleaning records today.",
+		"Oh, another day, another opportunity to clean up after you.",
+		"Who needs an armory when we have the Roomba to secure your guns in its dustbin?",
+		"Are you perhaps unfamiliar with proper ship etiquette, my dear crayon-eating passengers?",
+	)
+	activation_animation = "cleanbot-on"
+	deactivation_animation = "cleanbot-off"
+	alter_operating_mode = TRUE
+	active_icon_state = "cleanbot1"
+
+/obj/machinery/bot/cleanbot/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(suck_items))
+
+/obj/machinery/bot/cleanbot/process()
+	///holds the list of floors and dirs we use to evaluate which direction we're heading
+	var/list/dirtyfloors = list()
+	///used to hold what tile in the immediate area is dirtiest
+	var/highestdirtvalue = 0
+	///used to hold the dirt value of a given tile during evaluation, which is compared with highestdirtvalue
+	var/sampledirtvalue = 0
+	///used to override destination if we detect dirt or grime nearby
+	var/destdir = null
+	///controls what direction we're heading, is overridden by destdir if it has a value
+	var/newdir
+	for(var/dirn in GLOB.alldirs)
```

Comment:
Should this be cardinals only?

Reply:
Cleanbots can move in any direction (unlike Roombas), it's an intentional choice.